wikiclue/src/routes/change-username/+page.svelte

```diff
@@ -17,18 +37,33 @@
       return;
     }
     if (newUsername === $currentUsername) {
-      errorMessage = "Cannot enter the same username";
+      errorMessage = "Cannot enter the same username.";
+      return;
+    }
+    if(!await authHandlers.ensureUniqueUsername(newUsername)){
+      errorMessage = "Another user is using this name.";
       return;
     }
-    // More error checking that we decide on for usernames
 
-    currentUsername.set(newUsername);
     changeWasClicked.set(true);
   }
 
-  function okClicked(){
+  let successMessage = "";
+
+  async function okClicked(){
     // Would actually update the database here
-    currentUsername.set(newUsername);
+    try {
+      if(await authHandlers.updateUsername($currentUsername, newUsername)){
+        successMessage = "Username changed!"
+      }
+      else {
+        errorMessage = "Could not change username."
+      }
+    }
+    catch(error){
+        errorMessage = error;
+    }
+
     setTimeout(() => goto('/home'), 0);
```

Comment:
Can we stay on the change username page instead of routing to home after a successful change? Otherwise the use never sees the "username changed" message